consensus/hotstuff/cruisectl/block_time_controller.go

```diff
@@ -462,16 +448,35 @@ func (ctl *BlockTimeController) measureViewDuration(tb TimedBlock) error {
 	return nil
 }
 
-// processEpochSetupPhaseStarted processes EpochSetupPhaseStarted events from the protocol state.
+// processEpochExtended processes the EpochExtended protocol data.
+// Whenever we encounter an epoch extension, we:
+//   - set epochFallbackTriggered to true if this is the first extension encountered.
+//   - update the curr epoch final view and target end time with the extension data.
+//
+// No errors are expected during normal operation.
+func (ctl *BlockTimeController) processEpochExtended(_ uint64, _ *flow.Header, extension flow.EpochExtension) error {
+	if !ctl.epochFallbackTriggered {
+		// set epochFallbackTriggered to true when we encounter the first extension
+		ctl.epochFallbackTriggered = true
+	}
+	ctl.curEpochFinalView = extension.FinalView
```

Comment:
I would suggest adding sanity checks on these view values. There is something very similar [in the Consensus Committee](https://github.com/onflow/flow-go/pull/6154/files#diff-7236d580d88b376c2040caf16dec5857f2c23366052f90e6f161d6d4646ca5b3R52-R59), I think we can use the same checks.

Reply:
https://github.com/onflow/flow-go/pull/6102/commits/cc6810ab137fe0f0aa5ece2f8d8272dab692e06c